AWMScontroller.ino

```diff
@@ -0,0 +1,158 @@
+#include <SPI.h>              // include libraries
+#include <LoRa.h>
+
+const long frequency = 433E6;  // LoRa Frequency
+
+const int csPin = 15;          // LoRa radio chip select
+const int resetPin = 16;        // LoRa radio reset
+const int irqPin = 4;          // change for your board; must be a hardware interrupt pin
+int loraval;
+bool isPktAvailable = false;
+bool isTxdone = true;
+unsigned char localAddress = 0xff;     // address of this device
+unsigned char destination = 0x80;      // destination to send to
+byte msgCount = 0;
+String message;
+String device_id;
+String Lora_Ra02;
+int pos1, pos2;
+String incoming;
+
+void setup() {
+  Serial.begin(9600);                   // initialize serial
+
+  LoRa.setPins(csPin, resetPin, irqPin);
+
+  if (!LoRa.begin(frequency)) {
+    Serial.println("LoRa init failed. Check your connections.");
+    while (true);                       // if failed, do nothing
+  }
+
+  Serial.println("LoRa init succeeded.");
+  Serial.println();
+  Serial.println("LoRa Simple Gateway");
+  Serial.println("Only receive messages from nodes");
+  Serial.println("Tx: invertIQ enable");
+  Serial.println("Rx: invertIQ disable");
+  Serial.println();
+
+  LoRa.onReceive(onReceive);
+  LoRa.onTxDone(onTxDone);
+  LoRa_rxMode();
+}
+
+void LoRa_rxMode() {
+  LoRa.disableInvertIQ();               // normal mode
+  LoRa.receive();                       // set receive mode
+}
+
+void LoRa_txMode() {
+  LoRa.idle();                          // set standby mode
+  LoRa.enableInvertIQ();                // active invert I and Q signalsLoRa.idle(); 
+  //delay(5);
+}
+
+void txloop()
+{
+  if(isTxdone)
+  {
+   LoRa_rxMode();
+    isTxdone = false;
+  }
+   
+  if(!isPktAvailable)
+    return;
+    
+ 
+  isPktAvailable = false;
+  delay(1000);
+  LoRa_sendMessage("HELLO");

```

Comment:
what is this message

Reply:
code under test.

Reply:
please be careful about such mistakes going forward